db/range_del_aggregator.cc

```diff
@@ -115,6 +133,159 @@ Status RangeDelAggregator::AddTombstones(
   return Status::OK();
 }
 
+Status RangeDelAggregator::AddTombstone(RangeTombstone tombstone) {
+  auto& tombstone_map = GetTombstoneMap(tombstone.seq_);
+  if (collapse_deletions_) {
+    // In collapsed mode, we only fill the seq_ field in the TombstoneMap's
+    // values. The end_key is unneeded because we assume the tombstone extends
+    // until the next tombstone starts. For gaps between real tombstones and
+    // for the last real tombstone, we denote end keys by inserting fake
+    // tombstones with sequence number zero.
+    std::vector<RangeTombstone> new_range_dels{
```

Comment:
one of the motivations for using a vector of new points is that, in a future optimization, we're considering merging multiple new points simultaneously (e.g., all points from a snapshot stripe in an SST file that we just read), which should be faster than merging new points one-at-a-time.